mypy/suggestions.py

```diff
@@ -476,14 +478,20 @@ def find_node_by_file_and_line(self, file: str, line: int) -> Tuple[str, SymbolN
         # We must be sure about any edits in this file as this might affect the line numbers.
         tree = self.ensure_loaded(self.fgmanager.graph[modname], force=True)
         node = None  # type: Optional[SymbolNode]
+        closest_line = None
         for _, sym, _ in tree.local_definitions():
-            if isinstance(sym.node, FuncDef) and sym.node.line == line:
-                node = sym.node
-                break
-            elif isinstance(sym.node, Decorator) and sym.node.func.line == line:
-                node = sym.node
-                break
+            if isinstance(sym.node, FuncDef):
+                sym_line = sym.node.line
+            elif isinstance(sym.node, Decorator):
+                sym_line = sym.node.func.line
             # TODO: add support for OverloadedFuncDef.
+            else:
+                continue
+
+            # We want the closest function above the specified line
+            if sym_line <= line and (closest_line is None or sym_line > closest_line):
+                closest_line = sym_line
+                node = sym.node
```

Comment:
This has one downside, nested functions. I mean currently it doesn't work either, but now it will instead add an annotation to enclosing function (that may be already annotated). I have hit this several times last week.

I am not sure what to do here. If there is a simple way to support nested functions, then it would make sense to add it. Otherwise maybe open an issue or add a TODO here.